test/e2e/upgrade_test.go

```diff
@@ -41,6 +43,18 @@ func upgradeTestSuite(t *testing.T) {
 	require.NoError(t, err, "error testing Windows workloads")
 	defer testCtx.deleteJob(testerJob.Name)
 
+	// check BYOH enabled
+	if gc.numberOfBYOHNodes >= 1 {
```

Comment:
consider adding a require to check number of byohnodes is 1 instead of if statement to make sure we are always deploying workloads.

Reply:
As per our review session, that's a good point. I feel the step of actually enforcing as a hard requirement the presence of at least 1 BYOH node in the e2e-upgrade tests should takes place elsewhere.

The question is: Must the e2e-upgrade test run with at least 1 BYOH node?

Reply:
I think it should, how else would we ensure this particular scenario of having a pod and upgrading is tested?

Reply:
What we could do is break this into a separate test function and then lead it off with a check+skip as we do in other places. Example: https://github.com/openshift/windows-machine-config-operator/blob/6ba37ed99a01fdebf124c5098b02b0dffc6537f0/test/e2e/reconfigure_test.go#L65-L68

Reply:
What would be the specific test to skip, if no BYOH node is present?

Reply:
I think the suggestion is to skip the deployment on byoh node.